app/__init__.py

```diff
@@ -0,0 +1,33 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100, hidden: bool = False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    @property
+    def is_alive(self) -> bool:
+        return self.health > 0
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+    def bitten(self) -> None:
+        if self.hidden:
+            return
+
+        self.health -= 50
+        if not self.is_alive:
+            self.alive.remove(self)
+
+
+class Carnivore(Animal):
+    def bite(self, animal: Herbivore):
```

Comment:
Code Style issue: Use annotation, it is a good practice. So the `bite` method should be annotated with the return type `-> None`.